e2e/poms/task1.ts

```diff
@@ -0,0 +1,190 @@
+
+    async publishForm() {
+        await this.page.getByTestId(TASK1SELECTOR.testIds.publishButton).click();
+        await this.page.getByTestId(TASK1SELECTOR.testIds.moreDropdownIcon).click();
+        await this.page.getByTestId(TASK1SELECTOR.testIds.analyticsMoreTab).click();
+        await expect(
+            this.page.getByTestId(TASK1SELECTOR.testIds.visitsMetric).getByTestId(TASK1SELECTOR.testIds.insightsCount)
+        ).toHaveText("0");
+        await expect(
+            this.page.getByTestId(TASK1SELECTOR.testIds.startsMetric).getByTestId(TASK1SELECTOR.testIds.insightsCount)
+        ).toHaveText("0");
+        await expect(
+            this.page.getByTestId(TASK1SELECTOR.testIds.submissionsMetric).getByTestId(TASK1SELECTOR.testIds.insightsCount)
+        ).toHaveText("0");
+        await expect(
+            this.page.getByTestId(TASK1SELECTOR.testIds.completionRateMetric).getByTestId(TASK1SELECTOR.testIds.insightsCount)
+        ).toHaveText("0%");
+    }
+
+    async afterVisit() {
+        const initialVisitCount = Number(
+            await this.page.getByTestId(TASK1SELECTOR.testIds.visitsMetric).getByTestId(TASK1SELECTOR.testIds.insightsCount).textContent()
+        );
+        const page1Promise = this.page.waitForEvent("popup");
+        await this.page.getByTestId(TASK1SELECTOR.testIds.publishPreviewButton).click();
+        const page1 = await page1Promise;
+        await this.page.waitForTimeout(TASK1SELECTOR.timeouts.medium);
+        await this.page.reload();
+        await expect(
+            this.page.getByTestId(TASK1SELECTOR.testIds.visitsMetric).getByTestId(TASK1SELECTOR.testIds.insightsCount)
+        ).toHaveText(String(initialVisitCount + 1));
+        await page1.close();
+    }
+
+    async afterStart() {
+        const initialStartCount = Number(
+            await this.page.getByTestId(TASK1SELECTOR.testIds.startsMetric).getByTestId(TASK1SELECTOR.testIds.insightsCount).textContent()
+        );
+        const initialVisitCount = Number(
+            await this.page.getByTestId(TASK1SELECTOR.testIds.visitsMetric).getByTestId(TASK1SELECTOR.testIds.insightsCount).textContent()
+        );
+        const page2Promise = this.page.waitForEvent("popup");
+        await this.page.getByTestId(TASK1SELECTOR.testIds.publishPreviewButton).click();
+        const page2 = await page2Promise;
+        await page2.getByTestId(TASK1SELECTOR.testIds.emailTextField).fill(TASK1TEXT.texts.email);
+        await this.page.waitForTimeout(TASK1SELECTOR.timeouts.medium);
+        await this.page.reload();
+        await expect(
+            this.page.getByTestId(TASK1SELECTOR.testIds.visitsMetric).getByTestId(TASK1SELECTOR.testIds.insightsCount)
+        ).toHaveText(String(initialVisitCount + 1));
+        await expect(
+            this.page.getByTestId(TASK1SELECTOR.testIds.startsMetric).getByTestId(TASK1SELECTOR.testIds.insightsCount)
+        ).toHaveText(String(initialStartCount + 1));
+        return await page2.url();
```

Comment:
This is a redundant await. You can simply return the page2.url and the await statement for the method call will await this promise.

```suggestion
        return page2.url();
```